ent/encryption/encryption.cc

```diff
@@ -661,6 +661,7 @@ class encryption_file_io_extension : public sstables::file_io_extension {
                             sstables::component_type::TemporaryStatistics,
                             sstables::component_type::Partitions,
```

Comment:
> in advance that the partition key count estimate is good enough.
(Which should be the case in flushes and compactions).

The key estimate isn't very accurate during compaction. The output key count can range from 1 to the sum of the partition key counts of all the inputs.

Moreover, in incremental compaction, only subranges of the inputs go into the outputs.

Even memtable flushes can overestimate (if we compact-during-flush). This might be useful when flushing system.paxos or keyspace.user_table$paxos (those tables have many tombstones).

I think it's okay to write it unconditionally.